### PR TITLE
travis: Properly handle commits pushed/merged to master

### PR DESCRIPTION
**Pull request by jigpu**
Pushing or merging a branch to master updates the repository's reference
to that branch. When our travis script fetches the master branch and calls
`git format-patch` to produce a list of commits that aren't already on
master, there is obviously nothing to generate. This isn't a problem on
its own, but checkpatch the errors out with:

scripts/checkpatch.pl: *.patch: open failed - No such file or directory

To work around this we now only run checkpatch if patch files were
generated. Because we should only be pushing commits that were already
tested, this shouldn't significantly change our coverage.

Signed-off-by: Jason Gerecke <jason.gerecke@wacom.com>

### CODE REVIEW

**Comment by skomra** [comment]
Is this build below failing because of warnings?

https://travis-ci.org/skomra/input-wacom/builds/569584206?utm_source=github_status&utm_medium=notification

**Comment by jigpu** [comment]
Yeah. If you click on any of the jobs (e.g. 555.1) you can see the build output of that particular configuration. In all cases it looks like the build itself completes without a hitch (`The command "./autogen.sh [...] exited with 0.`) but checkpatch has a number of problems with the second patch in particular:

~~~
---------------------------------------------------------------

0001-travis-Properly-handle-commits-pushed-merged-to-mast.patch

---------------------------------------------------------------

total: 0 errors, 0 warnings, 5 lines checked

0001-travis-Properly-handle-commits-pushed-merged-to-mast.patch has no obvious style problems and is ready for submission.

--------------------------------------------------------

0002-HID-wacom-support-named-keys-on-older-devices.patch

--------------------------------------------------------

WARNING: line over 80 characters

#28: FILE: 4.5/wacom_wac.c:494:

+	bool wrench = false, keyboard = false, mute_touch = false, menu = false, info = false;

WARNING: line over 80 characters

#50: FILE: 4.5/wacom_wac.c:531:

+				wacom->shared->is_touch_on = !wacom->shared->is_touch_on;

WARNING: line over 80 characters

#64: FILE: 4.5/wacom_wac.c:545:

+				wacom->shared->is_touch_on = !wacom->shared->is_touch_on;

WARNING: line over 80 characters

#92: FILE: 4.5/wacom_wac.c:623:

+				    SW_MUTE_DEVICE, !wacom->shared->is_touch_on);

WARNING: suspect code indent for conditional statements (8, 24)

#103: FILE: 4.5/wacom_wac.c:1529:

+	if (wacom->shared->has_mute_touch_switch &&

[...]

+			if (!wacom->shared->touch_down)

total: 0 errors, 5 warnings, 144 lines checked

0002-HID-wacom-support-named-keys-on-older-devices.patch has style problems, please review.

---------------------------------------------------------------

0003-travis-Disable-warnings-about-duplicate-Signed-off-b.patch

---------------------------------------------------------------

total: 0 errors, 0 warnings, 7 lines checked

0003-travis-Disable-warnings-about-duplicate-Signed-off-b.patch has no obvious style problems and is ready for submission.
~~~

**Comment by skomra** [comment]
Sorry I was rushing when I wrote my question, let me rephrase:

 Are we going to fail our builds because of warnings? Probably all of the warnings above can be removed, but I'm not sure we can always avoid warnings.

There are a number of times I've had to send patches with warnings to the list to keep the style consistent with the surrounding code. Also sometimes the depth of the nested blocks requires making the line go over 80 characters.

I don't think the kernel will automatically reject a patch just because it has a warning. But maybe things have changed recently. If they have, future patches may require refactor patches just to satisfy checkpatch.

**Comment by jigpu** [comment]
That's a good point. On the one hand, I'd like to be notified when there are warnings present, but our history of not using checkpatch means that its going to be pretty noisy for the time being. I ran checkpatch on the upstream code and grabbed the top 10 warnings/errors and sorted by number of occurrences:

~~~
    186 WARNING: line over 80 characters
    155 ERROR: that open brace { should be on the previous line
     48 WARNING: struct wacom_features should normally be const
     23 WARNING: Prefer 'unsigned int' to bare use of 'unsigned'
     17 ERROR: else should follow close brace '}'
     14 ERROR: code indent should use tabs where possible
      8 WARNING: quoted string split across lines
      7 WARNING: Missing a blank line after declarations
      6 WARNING: please, no spaces at the start of a line
      5 WARNING: braces {} are not necessary for single statement blocks
~~~

Limiting the line-length warnings to those with 100+ chars causes the number of warnings to drop to just 6. The 155 open brace errors all appear to be entirely due to the style we use in the list of wacom_features structs. Everything else seems like reasonable things to fix up on the fly if we come across them.

Do you think of adding `--max-line-length=100 --ignore OPEN_BRACE` to our checkpatch command should suffice?

(EDIT: It looks like there are only 39 lines with 90+ chars if we want to try to be a bit more strict)

**Comment by skomra** [comment]
In the 4.5 path it looks like there's only 5 lines over 90 chars, so 90 seems fine.

**Review by skomra — APPROVED** [review]
Reviewed-by: Aaron Armstrong Skomra <aaron.skomra@wacom.com>
